symplyphysics/laws/thermodynamics/maxwell_boltzmann_statistics/speed_distribution.py

```diff
@@ -41,6 +54,63 @@
     * exp(-1 * particle_mass * particle_speed**2 / (2 * units.boltzmann_constant * equilibrium_temperature))
 )
 
+# Derive from Maxwell-Boltzmann distribution of velocity vector components
+
+_velocity_component_distribution = velocity_component_distribution.law.rhs.subs({
+    velocity_component_distribution.particle_mass: particle_mass,
+    velocity_component_distribution.equilibrium_temperature: equilibrium_temperature,
+})
+
+_velocity_x, _velocity_y, _velocity_z = sym_symbols("velocity_x:z", real=True)
+
+_velocity_x_distribution = _velocity_component_distribution.subs(
+    velocity_component_distribution.velocity_component, _velocity_x
+)
+
+_velocity_y_distribution = _velocity_component_distribution.subs(
+    velocity_component_distribution.velocity_component, _velocity_y
+)
+
+_velocity_z_distribution = _velocity_component_distribution.subs(
+    velocity_component_distribution.velocity_component, _velocity_z
+)
+
+_spherical_coordinate_system = CoordinateSystem(CoordinateSystem.System.SPHERICAL)
+
+_radius, _azimuthal_angle, _polar_angle = _spherical_coordinate_system.coord_system.base_scalars()
+
+# The speed distribution depends solely on the radial component of the velocity vector.
+# Therefore we integrate over polar and azimuthal angles to get rid of them.
+# We work in the three-dimensional velocity space `d^3(v)` so radius is speed (magnitude of velocity vector)
+_spherical_volume_velocity_element = (
+    sympify(volume_element_magnitude(_spherical_coordinate_system))
+    .subs(_radius, particle_speed)
+    .integrate((_polar_angle, 0, pi), (_azimuthal_angle, 0, 2*pi))
+)
+
+# The first three terms arise from the distribution of the velocity vector in Cartesian coordinates
+# and in order to switch to spherical coordinates we multiply by the spherical volume element of the
+# 3D velocity space found above.
```

Comment:
I am not convinced. I do not see how the multiplication of X, Y, Z components to volume element changes to spherical coordinates. Do you have any link to this proof source?

Reply:
![image](https://github.com/blackyblack/symplyphysics/assets/115647936/c1e3a8eb-df1c-477b-b3e4-5205bc1e7311)

Reply:
I understand it. What is the step explaining that `speed_distribution_function(particle_speed) == f(v_x) * f(v_y) * f(v_z) * dV` ?

Reply:
First we get the distribution function of the velocity vector in Cartesian coordinates $`f_{\vec v}(\vec v)`$ by multiplying the distribution functions of each of the velocity component since they are independent random variables (first sentence of the note), so we have $`f_{\vec v}(\vec v) d^3 \vec v = f_{v_x}(v_x) f_{v_y}(v_y) f_{v_z}(v_z) d^3 \vec v`$, where $`d^3 \vec v = dv_x dv_y dv_z`$. Afterwards we perform a substitution of variables $`(v_x, v_y, v_z) \to (v, \theta_v, \phi_v)`$ and the Jacobian appears (second sentence of the note).

Reply:
Add to description that distribution function of the vector is multiplication of distribution functions of each of the velocity component (that's where I was stuck) and we're good to go. You can also add a link to some related lemma if you have it available.

Reply:
Technically the Maxwell distribution initially has the formula for $`f_{\vec v}(\vec v)`$ and only then we get the distributions of the individual components of the velocity vector by integrating over the other two velocity components. Maybe it's worth making a separate law for $`f_{\vec v}(\vec v) d^3 \vec v`$, derive the individual distributions from it, and this proof would only feature the Jacobian and the change of variables.

Reply:
Interesting idea. You can try on it in another PR.